src/test/kotlin/MicronautAuthenticationTest.kt

```diff
@@ -0,0 +1,213 @@
+package com.cjbooms.fabrikt.generators
+
+import com.cjbooms.fabrikt.cli.CodeGenerationType
+import com.cjbooms.fabrikt.cli.ControllerCodeGenTargetType
+import com.cjbooms.fabrikt.configurations.Packages
+import com.cjbooms.fabrikt.generators.controller.MicronautControllerInterfaceGenerator
+import com.cjbooms.fabrikt.model.SourceApi
+import com.cjbooms.fabrikt.util.ResourceHelper.readTextResource
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.TypeSpec
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class MicronautAuthenticationTest {
+    private val basePackage = "authenticationTest"
+    private lateinit var generated: Collection<FileSpec>
+
+    @Suppress("unused")
+    private fun testCases(): Stream<Pair<String, String>> = Stream.of(
+        Pair("global_authentication_required.yml", "SecurityRule.IS_AUTHENTICATED"),
+        Pair("global_authentication_prohibited.yml", "SecurityRule.IS_ANONYMOUS"),
+        Pair("global_authentication_optional.yml", "SecurityRule.IS_AUTHENTICATED, SecurityRule.IS_ANONYMOUS"),
+    )
+
+    private fun setupTest(testPath: String): Collection<FileSpec> {
+        val api = SourceApi(readTextResource("/authenticationTest/$testPath"))
+        return MicronautControllerInterfaceGenerator(Packages(basePackage), api).generate().files
+    }
+
+    @BeforeEach
+    fun init() {
+        MutableSettings.updateSettings(
+            genTypes = setOf(CodeGenerationType.CONTROLLERS),
+            controllerTarget = ControllerCodeGenTargetType.MICRONAUT
+        )
+    }
+
+
+    // global authentication tests
+    @ParameterizedTest
+    @MethodSource("testCases")
+    fun `ensure that global authentication set to any authentication will add the required parameter`(testCase: Pair<String, String>) {
+        val controller = setupTest(testCase.first)
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
```

Comment:
**nitpick:**  `functionParameters` (the same everywhere else)

Reply:
done